h2gis-functions/src/main/java/org/h2gis/functions/io/geojson/GeoJsonReaderDriver.java

```diff
@@ -1329,14 +1338,18 @@ private void parseArrayMetadata(JsonParser jp) throws IOException {
      * @param jp the json parser
      * @return the object but written like a String
      */
-    private void parseObjectMetadata(JsonParser jp) throws IOException {
-        JsonToken value;
-        while(jp.nextToken() != JsonToken.END_OBJECT) {
-            value = jp.nextToken();
-            if (value == JsonToken.START_OBJECT) {
-                parseObjectMetadata(jp);
-            } else if (value == JsonToken.START_ARRAY) {
-                parseArrayMetadata(jp);
+    private void parseObjectMetadata(JsonParser jp, boolean notNull) throws IOException {
```

Comment:
Why did you add the notNull boolean ? In you function, if notNull is true, it doesn't do anything, so why not test in null or not and if not null, call the function ?